pkg/volume/flexvolume/plugin.go

```
@@ -0,0 +1,191 @@
+		spec:               spec,
+		readOnly:           readOnly,
+		blockDeviceMounter: &mount.SafeFormatAndMount{Interface: mounter, Runner: runner},
+	}, nil
+}
+
+// NewUnmounter is part of the volume.VolumePlugin interface.
+func (plugin *flexVolumePlugin) NewUnmounter(volName string, podUID types.UID) (volume.Unmounter, error) {
+	return plugin.newUnmounterInternal(volName, podUID, plugin.host.GetMounter(), plugin.runner)
+}
+
+// newUnmounterInternal is the internal unmounter routine to clean the volume.
+func (plugin *flexVolumePlugin) newUnmounterInternal(volName string, podUID types.UID, mounter mount.Interface, runner exec.Interface) (volume.Unmounter, error) {
+	return &flexVolumeUnmounter{
+		flexVolume: &flexVolume{
+			driverName: plugin.driverName,
+			execPath:   plugin.getExecutable(),
+			mounter:    mounter,
+			plugin:     plugin,
+			podUID:     podUID,
+			volName:    volName,
+		},
+		runner: runner,
+	}, nil
+}
+
+// NewAttacher is part of the volume.AttachableVolumePlugin interface.
+func (plugin *flexVolumePlugin) NewAttacher() (volume.Attacher, error) {
+	return &flexVolumeAttacher{plugin}, nil
+}
+
+// NewDetacher is part of the volume.AttachableVolumePlugin interface.
+func (plugin *flexVolumePlugin) NewDetacher() (volume.Detacher, error) {
+	return &flexVolumeDetacher{plugin}, nil
+}
+
+// ConstructVolumeSpec is part of the volume.AttachableVolumePlugin interface.
+func (plugin *flexVolumePlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.Spec, error) {
+	flexVolume := &api.Volume{
+		Name: volumeName,
+		VolumeSource: api.VolumeSource{
+			FlexVolume: &api.FlexVolumeSource{
+				Driver: plugin.driverName,
+			},
+		},
+	}
+	return volume.NewSpecFromVolume(flexVolume), nil
+}
+
+// Mark the given commands as unsupported.
+func (plugin *flexVolumePlugin) unsupported(commands ...string) {
+	plugin.Lock()
+	defer plugin.Unlock()
+	plugin.unsupportedCommands = append(plugin.unsupportedCommands, commands...)
```

Trying to understand this. Is the purpose here to have a cache, so that you can fail fast based on previous invocations of driver, instead of having to invoke the driver and parse the response every time?

Also, does the `plugin` object  have a separate instance for every driver? How is that controlled?